lib/transaction/pool.go

```diff
@@ -14,9 +19,17 @@ type Pool struct {
 
 // NewPool returns a new empty Pool
 func NewPool() *Pool {
-	return &Pool{
+	p := &Pool{
 		transactions: make(map[common.Hash]*ValidTransaction),
 	}
+
+	go metrics.CollectGaugeMetrics(
```

Comment:
I think it would be nicer if this was called outside of the `NewPool` constructor, and we could have some polling process to iterate over all the `GaugeCollector` implementations, rather than having multiple goroutines each polling their own gauge.

Reply:
Got it! I've implemented a `Collector`, that acts like a harvester, which has an `AddGauge` and a `Start` methods, then the `AddGauge` will "register" all the `GaugeMetrics` implementations so when `Start` is called (when the node starts) it triggers a goroutine that does polling in all the registereds implementations.

I think with this way we can remove all the spread goroutines and centralize in one in the `metrics` package.